components/eam/src/physics/rrtmgp/radiation.F90

```diff
@@ -413,14 +441,71 @@ end function radiation_nextsw_cday
 
    !================================================================================================
 
-   subroutine radiation_init(state)
+   subroutine get_aerosol_optical_property_from_file(month_int,aerosol_optical_property_str,band_identifier,nbands,aerosol_optical_property)
+      use pio,            only: file_desc_t, pio_nowrite
+      use cam_pio_utils,    only: cam_pio_openfile,cam_pio_closefile
+      use cam_grid_support, only: cam_grid_check,cam_grid_id,cam_grid_get_dim_names
+      use ncdio_atm,       only: infld
+      use time_manager,   only: get_curr_date
+      use cam_logfile,     only: iulog
+
+      integer, intent(in) :: month_int, nbands !nbands is either nswbands or nlwbands
+      character*(*), intent(in) :: aerosol_optical_property_str, band_identifier
+      real(r8), intent(inout) :: aerosol_optical_property(pcols,pver,nbands,begchunk:endchunk)
+      !real(r8), intent(inout) :: aerosol_optical_property(pcol,pver,nbands)
+
+     !internal variables
+      character(len=100) :: base_file_name, optics_lookup_dir
+      character(len=500) :: filename
+      character(len=20) :: dim1name, dim2name
+      character(len=20) :: mon_str
+      type(file_desc_t) :: nccn_ncid
+      integer :: year, month, day, tod, next_month, grid_id
+      logical :: found = .false.
+
+      write(mon_str,*) month_int
+
+      !assign base_file_name the name of the CCN file being used 
+      base_file_name = "spa_optics_file_"
+      optics_lookup_dir = "/p/lustre2/beydoun1/aerosol_optics"
+
+      mon_str = adjustl(mon_str)
+
+     !retrieve the name of the relevant file by combining base file name !with
+     !!month and full file path:
+      filename = trim(optics_lookup_dir)//'/'//trim(base_file_name)//trim(mon_str)//'.nc'
+
+      write(iulog,*) 'file name is', filename
```

Comment:
these writeouts are probably not needed once debugging is over. If they are permanent, you should be more clear about what they are. E.g. "file name" => "Prescribed aerosol file name"...